operator/cmd/operator/app/operator.go

```diff
@@ -110,6 +114,28 @@ func Run(ctx context.Context, o *options.Options) error {
 		return err
 	}
 
+	if err := manager.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		klog.Errorf("Unable to set up ready check: %v", err)
+		return err
+	}
+
+	// Unregister default NewGoCollector
+	ctrlmetrics.Registry.Unregister(collectors.NewGoCollector())
+
+	ctrlmetrics.Registry.MustRegister(
+		collectors.NewGoCollector(
+			collectors.WithGoCollectorRuntimeMetrics(
+				collectors.MetricsGC,
+				collectors.MetricsScheduler,
+				collectors.MetricsMemory,
+				collectors.GoRuntimeMetricsRule{Matcher: regexp.MustCompile(`^/sync/.*`)},
+			),
+		),
+	)
+	ctrlmetrics.Registry.MustRegister(
+		prometheusversion.NewCollector("karmada_operator"),
+	)
```

Comment:
I bet this meric would be useful, but I doubt it works as expected, because it tries to get the version info from the [prometheus version package](https://github.com/karmada-io/karmada/blob/3f919e8c479b5aeb65f5852bd031a8d9ccaee25b/vendor/github.com/prometheus/common/version/info.go#L27) and we don't set version to that. Have you tested it?

Reply:
Thanks @RainbowMango 
Got it.

I miss `gflags` to add it.